tests/tests_pytorch/tuner/test_lr_finder.py

```diff
@@ -441,6 +441,43 @@ def test_if_lr_finder_callback_already_configured():
         trainer.tune(model)
 
 
+def test_lr_finder_callback_restarting(tmpdir):
+    """Test that `LearningRateFinder` does not set restarting=True when loading checkpoint."""
+
+    class MyBoringModel(BoringModel):
+        def __init__(self):
+            super().__init__()
+            self.learning_rate = 0.123
+
+        def configure_optimizers(self):
+            return torch.optim.SGD(self.parameters(), lr=self.learning_rate)
+
+    class CustomLearningRateFinder(LearningRateFinder):
+        milestones = (1,)
+
+        def lr_find(self, trainer, pl_module) -> None:
+            super().lr_find(trainer, pl_module)
+            assert not trainer.fit_loop.restarting
+
+        def on_train_epoch_start(self, trainer, pl_module):
+            if trainer.current_epoch in self.milestones or trainer.current_epoch == 0:
+                self.lr_find(trainer, pl_module)
+
+    model = MyBoringModel()
+    trainer = Trainer(
+        default_root_dir=tmpdir,
+        max_epochs=3,
+        callbacks=[CustomLearningRateFinder(early_stop_threshold=None, update_attr=True)],
```

Comment:
could you double check, is `early_stop_threshold=None` really a valid value? The type hint says no but I also remember I saw it in some examples.

Reply:
@awaelchli It is supported according to the underlying logic so I also updated the type annotations here.